cpp/src/parquet/CMakeLists.txt

```diff
@@ -260,6 +263,14 @@ endif()
 if(NOT PARQUET_MINIMAL_DEPENDENCY)
   list(APPEND PARQUET_SHARED_LINK_LIBS arrow_shared)
 
+  # Add RapidJSON libraries
+  if(NOT ARROW_JSON)
```

Comment:
Do we need to modify the CMake config to enable `ARROW_JSON` when `ARROW_PARQUET` is ON? cc @kou @raulcd

Reply:
We don't need JSON if `PARQUET_MINIMAL_DEPENDENCY` is `ON`, right?

Then, we don't need the following:

```diff
diff --git a/cpp/cmake_modules/DefineOptions.cmake b/cpp/cmake_modules/DefineOptions.cmake
index 16360929a9..93a7ba6063 100644
--- a/cpp/cmake_modules/DefineOptions.cmake
+++ b/cpp/cmake_modules/DefineOptions.cmake
@@ -370,7 +370,8 @@ takes precedence over ccache if a storage backend is configured" ON)
                 "Build the Parquet libraries"
                 OFF
                 DEPENDS
-                ARROW_IPC)
+                ARROW_IPC
+                ARROW_JSON)
 
   define_option(ARROW_ORC
                 "Build the Arrow ORC adapter"
```

Reply:
It seems that https://github.com/apache/parquet-cpp/pull/279 introduced it.

Reply:
Ha... so we've had this option for ~8 years and it was never tested or documented anywhere. I've opened https://github.com/apache/arrow/issues/46219 so that we may eventually remove it.

Reply:
> Do we need to modify the CMake config to enable `ARROW_JSON` when `ARROW_PARQUET` is ON?

`ARROW_JSON` will build the Arrow JSONL reader, which should not be required for Parquet. We just need RapidJSON here.

Reply:
I would suggest this diff:
```diff
diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index 1af347914f..92ea7e346f 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -374,16 +374,13 @@ target_include_directories(arrow::flatbuffers
 # ----------------------------------------------------------------------
 # Some EP's require other EP's
 
-if(PARQUET_REQUIRE_ENCRYPTION)
-  set(ARROW_JSON ON)
-endif()
-
 if(ARROW_WITH_OPENTELEMETRY)
   set(ARROW_WITH_NLOHMANN_JSON ON)
   set(ARROW_WITH_PROTOBUF ON)
 endif()
 
 if(ARROW_PARQUET)
+  set(ARROW_WITH_RAPIDJSON ON)
   set(ARROW_WITH_THRIFT ON)
 endif()
 
```

Reply:
FYI: We can remove this `if(NOT ARROW_JSON)` with the suggested change.